testsuite/MDAnalysisTests/guesser/test_default_guesser.py

```diff
@@ -0,0 +1,236 @@
+
+    def test_guess_masses_miss(self):
+        topology = Topology(2, attrs=[Atomtypes(['X', 'Z'])])
+        u = mda.Universe(topology)
+        u.guess_TopologyAttributes(to_guess=['masses'])
+        assert_allclose(u.atoms.masses, np.array([0.0, 0.0]), rtol=1e-3, atol=0)
+
+    @pytest.mark.parametrize('element, value', (('H', 1.008), ('XYZ', 0.0), ))
+    def test_get_atom_mass(self, element, value, default_guesser):
+        default_guesser.get_atom_mass(element) == approx(value)
+
+    def test_guess_atom_mass(self, default_guesser):
+        assert default_guesser.guess_atom_mass('1H') == approx(1.008)
+
+    def test_guess_masses_with_no_reference_elements(self):
+        u = mda.Universe.empty(3)
+        with pytest.raises(ValueError, match=('there is no reference attributes ')):
+            u.guess_TopologyAttributes('default', ['masses'])
+
+
+class TestGuessTypes(object):
+
+    def test_guess_types(self):
+        topology = Topology(2, attrs=[Atomnames(['MG2+', 'C12'])])
+        u = mda.Universe(topology)
+        u.guess_TopologyAttributes(to_guess=['types'])
+        values = DefaultGuesser(None).guess_types(u.atoms.names)
+        assert isinstance(u.atoms.types, np.ndarray)
+        assert_equal(u.atoms.types, np.array(['MG', 'C'], dtype=object))
+        assert_equal(values, np.array(['MG', 'C'], dtype=object))
+
+    def test_guess_atom_element(self, default_guesser):
+        assert default_guesser.guess_atom_element('MG2+') == 'MG'
+
+    def test_guess_atom_element_empty(self, default_guesser):
+        assert default_guesser.guess_atom_element('') == ''
+
+    def test_guess_atom_element_singledigit(self, default_guesser):
+        assert default_guesser.guess_atom_element('1') == '1'
+
+    def test_guess_atom_element_1H(self, default_guesser):
+        assert default_guesser.guess_atom_element('1H') == 'H'
+        assert default_guesser.guess_atom_element('2H') == 'H'
+
+    @pytest.mark.parametrize('name, element', (
+        ('AO5*', 'O'),
+        ('F-', 'F'),
+        ('HB1', 'H'),
+        ('OC2', 'O'),
+        ('1he2', 'H'),
+        ('3hg2', 'H'),
+        ('OH-', 'O'),
+        ('HO', 'H'),
+        ('he', 'H'),
+        ('zn', 'ZN'),
+        ('Ca2+', 'CA'),
+        ('CA', 'C'),
+    ))
+    def test_guess_element_from_name(self, name, element, default_guesser):
+        assert default_guesser.guess_atom_element(name) == element
+
+
+def test_guess_charge(default_guesser):
+    # this always returns 0.0
+    assert default_guesser.guess_atom_charge('this') == approx(0.0)
+
+
+def test_guess_bonds_Error():
+    u = make_Universe(trajectory=True)
+    with pytest.raises(ValueError):
+        u.guess_TopologyAttributes(to_guess=['bonds'])
+
+
+def test_guess_bond_coord_error():
+    u = mda.Universe(datafiles.PDB)
+    with pytest.raises(ValueError):
+        DefaultGuesser(None).guess_bonds(u.atoms, [[1,2,2]])
```

Comment:
```suggestion
        DefaultGuesser(u).guess_bonds(u.atoms, [[1,2,2]])
```

Why not pass the universe?